### ISSUE TITLE
Implement multiple buttons to the light_switch example (CON-1001)

### ISSUE THREAD

**Issue by johnvigg** [espressif/esp-matter#821]
**Describe the question/query that you have**
I am trying to add more buttons to the light_switch example and control multiple other ESP's running the light example

**Additional context**
I have 5 ESP32c6's and I want to add for example 3 more buttons to the ESP32c6 but the code is quite overwhelming for me. It would be helpful if you could point me to exactly where I can make the changes. 

In the end, there will be 3 more end points and each of these will be binded to one of the other ESP's

Thank you for your help in advance

**Comment by sayondeep** [comment]
@johnvigg Please see [here](https://github.com/espressif/esp-matter/blob/52a95eaacb9b2075697bb854880e0e7392e8e438/examples/generic_switch/main/app_main.cpp#L209) for how to add more buttons.

**Comment by dhrishi** [comment]
@johnvigg The discriminator and passcode values are different in case of generic_switch example. Please see this: https://github.com/espressif/esp-matter/tree/main/examples/generic_switch#1-additional-environment-setup

**Comment by johnvigg** [comment]
@dhrishi , as I mentioned when pairing I did use the descriminator and passcodes as mentioned in the Readme file. But when I write matter device factoryreset on the serial monitor, it comes up with the error as shown above and I think it is not broadcasting on bluetooth because when I run the pairing command on my commissioner, it says it could not find any devices.

**Comment by dhrishi** [comment]
> But when I write matter device factoryreset on the serial monitor, it comes up with the error as shown above and I think it is not broadcasting on bluetooth because when I run the pairing command on my commissioner, it says it could not find any devices.

This looks like an issue. We will check.

**Comment by shripad621git** [comment]
@johnvigg, in the generic switch example, the discriminator and passcode to commission the device is different because the `factory_partition` is being flashed at `0x10000` which contains the `discriminator `and `passcode` as `3841 `and `20202020` respectively.
Have you flashed the factory partititon specified in the [README](https://github.com/espressif/esp-matter/blob/main/examples/generic_switch/README.md) .
Please follow the steps below:
1.Flash the example using idf.py flash.
2.Please flash` esptool.py -p [port-name] write_flash 0x10000 mfg_binaries/20202020_3841.bin` after flashing the example.
3. Monitor the output using idf.py monitor.
4. Commission the device using 'chip-tool pairing ble-wifi 0x7283 [ssid] [password] 20202020 3841'.

I have tried the example with the above mentioned steps and it worked fine. Can you please try the above steps and confirm?

**Comment by johnvigg** [comment]
@shripad621git , it seems that I have overlooked this step, thank you for bringing it up. 

I am now able to commission the device now, but my original issue still question still persists. 
Again I have not made any changes to the code and I see that the the boot button in my ESP32C6 is setup already and I should be able to bind it to my other esp running the light example. but when I run the commands used in the light switch example, I get an error.
(10 - switch, 11 - light)
`accesscontrol write acl '[{"privilege": 5, "authMode": 2, "subjects": [ 112233, 10 ], "targets": null}]' 11 0x0` 
`binding write binding '[{"node":11, "endpoint":1, "cluster":6}]' 10 0x1` 
I get the error saying it is an unsupported cluster.

`
>>> binding write binding '[{"node":11, "endpoint":1, "cluster":6}]' 10 0x1
[1707227570.610297][28902:28902] CHIP:TOO: Command: binding write binding [{"node":11, "endpoint":1, "cluster":6}] 10 0x1 
[1707227570.611008][28902:28904] CHIP:TOO: Sending command to node 0xa
[1707227570.611566][28902:28904] CHIP:CSM: FindOrEstablishSession: PeerId = [1:000000000000000A]
[1707227570.611586][28902:28904] CHIP:CSM: FindOrEstablishSession: No existing OperationalSessionSetup instance found
[1707227570.611610][28902:28904] CHIP:DIS: Found an existing secure session to [1:000000000000000A]!
[1707227570.611626][28902:28904] CHIP:DIS: OperationalSessionSetup[1:000000000000000A]: State change 1 --> 5
[1707227570.611664][28902:28904] CHIP:TOO: 	cluster 0x0000_001E, attribute: 0x0000_0000, endpoint 1
[1707227570.611701][28902:28904] CHIP:DMG: WriteClient moving to [AddAttribu]
[1707227570.611724][28902:28904] CHIP:DMG: WriteClient moving to [AddAttribu]
[1707227570.611866][28902:28904] CHIP:EM: <<< [E:42849i S:15413 M:224939376] (S) Msg TX to 1:000000000000000A [314B] [UDP:[fe80::424c:caff:fe45:33dc%eno1]:5540] --- Type 0001:06 (IM:WriteRequest)
[1707227570.611973][28902:28904] CHIP:DMG: WriteClient moving to [AwaitingRe]
[1707227570.669557][28902:28904] CHIP:EM: >>> [E:42849i S:15413 M:35972043 (Ack:224939376)] (S) Msg RX from 1:000000000000000A [314B] --- Type 0001:07 (IM:WriteResponse)
[1707227570.669596][28902:28904] CHIP:EM: Found matching exchange: 42849i, Delegate: 0x7f724403b0e0
[1707227570.669624][28902:28904] CHIP:EM: Rxd Ack; Removing MessageCounter:224939376 from Retrans Table on exchange 42849i
[1707227570.669645][28902:28904] CHIP:DMG: WriteClient moving to [ResponseRe]
[1707227570.669684][28902:28904] CHIP:DMG: WriteResponseMessage =
[1707227570.669697][28902:28904] CHIP:DMG: {
[1707227570.669707][28902:28904] CHIP:DMG: 	AttributeStatusIBs =
[1707227570.669730][28902:28904] CHIP:DMG: 	[
[1707227570.669740][28902:28904] CHIP:DMG: 		AttributeStatusIB =
[1707227570.669754][28902:28904] CHIP:DMG: 		{
[1707227570.669765][28902:28904] CHIP:DMG: 			AttributePathIB =
[1707227570.669780][28902:28904] CHIP:DMG: 			{
[1707227570.669795][28902:28904] CHIP:DMG: 				Endpoint = 0x1,
[1707227570.669811][28902:28904] CHIP:DMG: 				Cluster = 0x1e,
[1707227570.669825][28902:28904] CHIP:DMG: 				Attribute = 0x0000_0000,
[1707227570.669837][28902:28904] CHIP:DMG: 			}
[1707227570.669858][28902:28904] CHIP:DMG: 				
[1707227570.669870][28902:28904] CHIP:DMG: 			StatusIB =
[1707227570.669883][28902:28904] CHIP:DMG: 			{
[1707227570.669895][28902:28904] CHIP:DMG: 				status = 0xc3 (UNSUPPORTED_CLUSTER),
[1707227570.669909][28902:28904] CHIP:DMG: 			},
[1707227570.669924][28902:28904] CHIP:DMG: 			
[1707227570.669937][28902:28904] CHIP:DMG: 		},
[1707227570.669962][28902:28904] CHIP:DMG: 		
[1707227570.669976][28902:28904] CHIP:DMG: 		AttributeStatusIB =
[1707227570.669993][28902:28904] CHIP:DMG: 		{
[1707227570.670006][28902:28904] CHIP:DMG: 			AttributePathIB =
[1707227570.670020][28902:28904] CHIP:DMG: 			{
[1707227570.670032][28902:28904] CHIP:DMG: 				Endpoint = 0x1,
[1707227570.670045][28902:28904] CHIP:DMG: 				Cluster = 0x1e,
[1707227570.670057][28902:28904] CHIP:DMG: 				Attribute = 0x0000_0000,
[1707227570.670070][28902:28904] CHIP:DMG: 				ListIndex = Null,
[1707227570.670081][28902:28904] CHIP:DMG: 			}
[1707227570.670097][28902:28904] CHIP:DMG: 				
[1707227570.670109][28902:28904] CHIP:DMG: 			StatusIB =
[1707227570.670121][28902:28904] CHIP:DMG: 			{
[1707227570.670133][28902:28904] CHIP:DMG: 				status = 0xc3 (UNSUPPORTED_CLUSTER),
[1707227570.670146][28902:28904] CHIP:DMG: 			},
[1707227570.670158][28902:28904] CHIP:DMG: 			
[1707227570.670169][28902:28904] CHIP:DMG: 		},
[1707227570.670186][28902:28904] CHIP:DMG: 		
[1707227570.670198][28902:28904] CHIP:DMG: 	],
[1707227570.670224][28902:28904] CHIP:DMG: 	
[1707227570.670235][28902:28904] CHIP:DMG: 	InteractionModelRevision = 11
[1707227570.670247][28902:28904] CHIP:DMG: }
[1707227570.670334][28902:28904] CHIP:DMG: WriteClient moving to [AwaitingDe]
[1707227570.670354][28902:28904] CHIP:TOO: Response Failure: IM Error 0x000005C3: General error: 0xc3 (UNSUPPORTED_CLUSTER)
[1707227570.670376][28902:28904] CHIP:TOO: Run command failure: IM Error 0x000005C3: General error: 0xc3 (UNSUPPORTED_CLUSTER)
[1707227570.670548][28902:28904] CHIP:EM: <<< [E:42849i S:15413 M:224939377 (Ack:35972043)] (S) Msg TX to 1:000000000000000A [314B] [UDP:[fe80::424c:caff:fe45:33dc%eno1]:5540] --- Type 0000:10 (SecureChannel:StandaloneAck)
[1707227570.670620][28902:28904] CHIP:EM: Flushed pending ack for MessageCounter:35972043 on exchange 42849i
>>> 
`
Should I make any changes before I can get this to work?

**Comment by shripad621git** [comment]
@johnvigg , you can follow the instructions mentioned [here](https://github.com/espressif/esp-matter/tree/main/examples/light_switch#21-bind-light-to-switch) for trying out the binding use-case.

**Comment by johnvigg** [comment]
@shripad621git , the commands I used are in fact from the light_switch example.

I wrote the acl values as mentioned in the guide to the light and then tried the binding command on the generic switch device and it gave the above error. I do not see any other information on the readme file regarding this.

I also tried a simple read command using the binding cluster and I got a similar error saying unsupported cluster. Does this mean the cluster is not supported?

`binding read binding 10 1`

`[1707231127.167505][48679:48687] CHIP:TOO: Response Failure: IM Error 0x000005C3: General error: 0xc3 (UNSUPPORTED_CLUSTER)
[1707231127.167567][48679:48687] CHIP:TOO: Run command failure: IM Error 0x000005C3: General error: 0xc3 (UNSUPPORTED_CLUSTER)
`

**Comment by shripad621git** [comment]
@johnvigg , you can tryout the binding functionality with light_switch example. The reason being
binding cluster is not present in `generic_switch `since the `generic switch` example has `generic_switch device type` and` light_switch example` has `on_off_switch device type.`  You can see the reference below 

-  [link1](https://github.com/espressif/esp-matter/blob/8a421d1f1823224c73b5c91f4ab277118212f2c2/components/esp_matter/esp_matter_endpoint.h#L212C11-L212C24)
-  [link2](https://github.com/espressif/esp-matter/blob/8a421d1f1823224c73b5c91f4ab277118212f2c2/components/esp_matter/esp_matter_endpoint.h#L251).

**Comment by johnvigg** [comment]
@shripad621git , thank you for the clarification.

I have already tried the light switch example and it worked well, but I wanted to add multiple buttons, so I looked into the generic switch example.

What would you recommend, modifying the light switch example or the generic switch one. I was able to easily add multiple buttons to the generic example. Is it possible to add the binding cluster functionality to this device type?

**Comment by johnvigg** [comment]
@sayondeep , if you recommend that method, I will try it out and get back to you.

**Comment by shripad621git** [comment]
> I have tried adding the binding cluster to the generic switch example. I was able to add that successfully. You can take the reference of the other clusters added in the generic switch example. For ex: fixed and user- label clusters

You can confirm that the addition of binding cluster is successful by 
`chip-tool descriptor read server-list <node-id> <endpoint-id of switch>`.You will get 30 as output in server-list which is the id for binding cluster. 
[Link](https://github.com/project-chip/connectedhomeip/blob/master/data_model%2Fclusters%2FBinding-Cluster.xml)

**Comment by johnvigg** [comment]
@shripad621git , I got it working by combining the code from the light switch example, like the client, binding and group related clusters. The only issue I am facing is adding 3 buttons, I have added 2 extra ones successfully but when I add the third one, and press it, I get the below message.

` 
I (603072) app_driver: Initial button pressed `
`I (603082) chip[ZCL]: 0x42136c78 ep ffff clus 0x0000_003B attr 0x0000_0001 not supported`
`I (603092) chip[ZCL]: SwitchServer: OnInitialPress
`

whereas the correctly configured buttons look like this

`I (1236372) app_driver: Initial button pressed`
`I (1236382) esp_matter_attribute: ********** W : Endpoint 0x0001's Cluster 0x0000003B's Attribute 0x00000001 is 1 **********`
`I (1236392) chip[ZCL]: SwitchServer: OnInitialPress
`

I also could not find where this error was called from, does anyone know why it gives an error? 
Is it something with not being able to create an endpoint?
